target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf289f.dts

```diff
@@ -0,0 +1,436 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+// Copyright (c) 2022, Pawel Dembicki <paweldembicki@gmail.com>.
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+        model = "ZTE MF289F";
+        compatible = "zte,mf289f";
+
+        aliases {
+                led-boot = &led_internal;
+                led-failsafe = &led_internal;
+                led-running = &led_internal;
+                led-upgrade = &led_internal;
+        };
+
+        chosen {
+                /*
+                 * bootargs forced by u-boot bootipq command:
+                 * 'ubi.mtd=rootfs root=mtd:ubi_rootfs rootfstype=squashfs rootwait'
+                 */
+                bootargs-append = " root=/dev/ubiblock0_1";
+        };
+
+        gpio-restart {
+                compatible = "gpio-restart";
+                gpios = <&tlmm 8 GPIO_ACTIVE_HIGH>;
+        };
+
+        leds {
+                compatible = "gpio-leds";
+
+                led_internal: led-0 {
+                        function = LED_FUNCTION_STATUS;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 10 GPIO_ACTIVE_LOW>;
+                        label = "blue:internal_led";
+                        default-state = "keep";
+                };
+
+                led-1 {
+                        function = LED_FUNCTION_WLAN;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 61 GPIO_ACTIVE_LOW>;
+                        linux,default-trigger = "phy0tpt";
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        linux,code = <KEY_WPS_BUTTON>;
+                        gpios = <&tlmm 68 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        soc {
+                rng@22000 {

```

Comment:
Most of these nodes have labels, so use those as handles instead of putting them manually in the soc node.

Reply:
To be clear, this comments applies on remaining nodes like MDIO, crypto etc.

Reply:
Can you explain better? I don't understand what to do..

Reply:
@stich86 look into the .dtsi hierarchy and search for nodes with same name, they are written as `<label>: name@address {...};`  - you can refer to them just by creating `&<label> {};` like for example, `&wifi0 {...};`, instead of extending the path.

Reply:
so looking at **qcom-ipq4019.dtsi** i've these duplicate nodes:

rng@22000 
mdio@9000 (but with different properties on this dts)
crypto@8e3a000
watchdog@b017000

so on 289F dts how I have to reference them?

for the three nodes with no properties (only status ok) i think the can be written in this way:

```
&watchdog {
                        status = "okay";
};
```

?

thanks

Reply:
Just look at the DTSI, they have labels before the node.
So just reference them like:
```
&mdio
&prng
&crypto (Dont forget to enable cryptobam as well)
&watchdog
```

Reply:
ok fixed, let me know if are ok now and can proceed with rebase